### ISSUE TITLE
Unable to replicate results after retraining

### ISSUE THREAD

**Issue by aemrey** [aimagelab/meshed-memory-transformer#5]
Hello and thank you for this fantastic repo!

I am trying to retrain your model using COCO features I have extracted myself using the [bottom-up attention repo](https://github.com/peteanderson80/bottom-up-attention/) as you have suggested in #2. I am currently on epoch 15 and the highest CIDEr score on the test set has been 1.13. This is much less than the 1.31 that I get when using your pretrained model. Other than the new features, I am using your default values for all hyperparameters.

Could you give me some guidance in order to better replicate your results?

**Comment by baraldilorenzo** [comment]
Dear @aemrey,

thanks for your interest in our code. From what you mention, it seems that the RL training part has not even started so it is fairly normal that you are getting lower results. Please keep it running until the training stops by itself.

Lorenzo.

**Comment by alesolano** [comment]
Hi @aemrey, I'm following your steps and right now I'm extracting the features of my set of images. I'm using [this model](https://github.com/peteanderson80/bottom-up-attention/blob/master/models/vg/ResNet-101/faster_rcnn_end2end_final/test.prototxt) loaded with [these weights](https://www.dropbox.com/s/wqada4qiv1dz9dk/resnet101_faster_rcnn_final.caffemodel?dl=1).

Now, how do you pack those features into an `Nx2048` tensor? I understood "features" by the output of the blob `cls_prob` [here](https://github.com/peteanderson80/bottom-up-attention/blob/master/models/vg/ResNet-101/faster_rcnn_end2end_final/test.prototxt#L7123), but it returns an `Nx1601` tensor. I'm sure I'm missing something here, maybe not taking the correct blob or using the wrong model.

Thanks!

P.S.: Don't know if I should maybe open a new issue

**Comment by ruotianluo** [comment]
I ran a couple of times your code.

<img width="245" alt="image" src="https://user-images.githubusercontent.com/16023153/76662386-cb8a1c00-654b-11ea-8a7e-4a3a1d5a20ac.png">


The test cider is always below 1.30. Any clues?

**Comment by marcellacornia** [comment]
Hi @ruotianluo, 
thanks for your interest in our work.

We noticed that the results can be different by changing the underlying architecture. For this reason, we also provided the weights of our final model. 

In our experiments, we used an NVIDIA 2080 Ti GPU. The other settings are the ones we reported in our repository.

**Comment by wanboyang** [comment]
> I ran a couple of times your code.
> 
> <img alt="image" width="245" src="https://user-images.githubusercontent.com/16023153/76662386-cb8a1c00-654b-11ea-8a7e-4a3a1d5a20ac.png">
> 
> The test cider is always below 1.30. Any clues?

I think it caused by the difference between bottom-up features provided in [1] and provided in this project. The testing results shown in ![image](https://github.com/wanboyang/image_dir/blob/main/QQ%E5%9B%BE%E7%89%8720201019142909.jpg), the curve named m2_transformer_wan used the features provided in [1] and the curve named m2_transformer_wan n used the features provided in this project.
I compare the features of [1] and this project , and find numbers of box in some image_id are different.
